packages/vite/src/node/config.ts

```diff
@@ -501,7 +499,7 @@ export async function resolveConfig(
     packageCache: new Map(),
     createResolver,
     optimizeDeps: {
-      disabled: config.optimizeDeps === false,
+      disabled: optimizeDeps.disabled === true,
```

Comment:
I added it so disabled will be always defined in the resolved config. Isn't that better than leaving the `undefined` when it is enabled?

Reply:
The resolved types is also `boolean | undefined`, so I think it's fine if it's `undefined` too, since we only check when `disabled` is truthy.

Reply:
With other options we are normally fully resolving them. Should we change the types in `ResolvedConfig` so it can't be `undefined`?

```ts
optimizeDeps: Omit<DepOptimizationOptions, 'keepNames' | 'disabled'> & { disabled: boolean }
```

Reply:
Removed the line for now, as we are not fully resolving other options in `optimizeDeps` either. We could have for v3 maybe a `ResolvedDepOptimizationOptions` so we can resolve all of them

Reply:
Yeah we could look into deeply resolving them in the future too. Thanks for making the change.